lib/public/views/Runs/Details/RunDetailsModel.js

```diff
@@ -76,6 +76,9 @@ export class RunDetailsModel extends Observable {
 
             this._fetchOneRun();
             this._fetchReasonTypes();
+        } else {
+            this._runNumber = runNumber;
+            this._fetchOneRun();
```

Comment:
When you end up with one line being the same in if and in else, often you should simply move it out of the if. The `_fetchReasonTypes` and `_fetchOneRun` are both asynchronous and you do not care about the order, so you can swap them. Same goes for the `this._runNumber = runNumber`. Then you see that you end both the if and else with the same lines (setting `this._runNumber` and `_fetchOneRun`) so you can move it after the if/else.

Another point, you have not put `this.runDetails = RemoteData.notAsked` probably because you see that this is overriden right after in `this._fetchOneRun`, I suggest that you remove it also from the if (same goes for the `this._eorReasonTypes = RemoteData.notAsked();`